### PR TITLE
Fix some flow definitions for AMInstallButton

### PR DESCRIPTION
**Pull request by willdurand**
I am working on a bigger patch that brings flow to disco pane. Doing this made me discover these issues.

### CODE REVIEW

File: src/core/components/AMInstallButton/index.js [hunk 1]
```diff
@@ -45,23 +45,23 @@ import type { ReactRouterLocationType } from 'core/types/router';
 import './styles.scss';
 
 type Props = {|
+  ...WithInstallHelpersInjectedProps,
   addon: AddonType,
   className?: string,
   defaultInstallSource: string,
   disabled: boolean,
-  location: ReactRouterLocationType,
```
Comment (rebmullin) [inline src/core/components/AMInstallButton/index.js]: should you setCurrentStatus that you have added above?

File: src/amo/pages/Addon/index.js [hunk 2]
```diff
@@ -535,6 +536,7 @@ export class AddonBase extends React.Component {
                       hasAddonManager={hasAddonManager}
                       install={install}
                       installTheme={installTheme}
+                      setCurrentStatus={setCurrentStatus}
```
Comment (rebmullin) [inline src/amo/pages/Addon/index.js]: I don't see this being used in the AMInstallButton - is that something you're adding in a later PR?

Comment (willdurand) [reply src/amo/pages/Addon/index.js]: nope, but it needs to be passed.

File: src/core/components/AMInstallButton/index.js (hunk 1 above)
Comment (willdurand) [reply src/core/components/AMInstallButton/index.js]: it is part of `WithInstallHelperInjectedProps`.

Comment (rebmullin) [reply src/core/components/AMInstallButton/index.js]: ahh..I see..cool 👍

**Pull request merged**